iOS Client/DevotionsController/MBDevotionsViewController.swift

```diff
@@ -64,5 +77,17 @@ class MBDevotionsViewController: UIViewController, StoreSubscriber, UITableViewD
     func numberOfSections(in tableView: UITableView) -> Int {
         return 1
     }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+       let devotion = devotions[indexPath.row]
+        MBStore.sharedStore.dispatch(SelectedDevotion(devotion: devotion))
+        do {
+            try store.saveDevotions(devotions: devotions)
```

Comment:
before this, where does the read property on the devotion actually get toggled to 'true' ?

Reply:
So that happens in the reducer during the SelectedDevotion action

Reply:
Hmm, I am OK with this for now, but I think it needs to be clearer in an upcoming refactor when we mark a devotion as read

The concerning thing is that the call to dispatch has a side effect of updating this view controller's devotions property that the next lines rely on.

I think it would be clearer to only do the dispatch here, then in `newState` check if the selected devotion in the devotions property is currently marked as false and, if so, call a helper to do the save

Reply:
I agree that it isn't ideal that the view controllers devotions rely on that dispatched action. Although, I don't think saving in the reducer is a good idea. Reducers shouldn't have any side effects like saving data or fetching data, they should just take a state and an action and return a new state, nothing else. I'll try to think of a better solution but might be good to just get this in.